src/crud.py

```diff
@@ -0,0 +1,81 @@
+from typing import Optional
+from sqlalchemy.orm import Session
+from . import models, schemas
+
+
+def get_user(db: Session, address: str):
+    return db.query(models.User).filter(models.User.address == address).first()
+
+def create_user(db: Session, user: schemas.UserCreation):
+    db_user = models.User(**user.model_dump())
+    db.add(db_user)
+    db.commit()
+    db.refresh(db_user)
+    return db_user
+
+def get_contracts_by_user(db: Session, user_address: str):
+    user = get_user(db, user_address)
+    if not user:
+        return None
+    return user.contracts
+
+
+def get_contract(db: Session, address: str):
+    return db.query(models.Contract).filter(models.Contract.address == address).first()
+
+
+def get_entrypoints(db: Session, contract_address: str):
+    contract = get_contract(db, contract_address)
+    return contract.entrypoints if contract else []
+
+def get_entrypoint(db: Session, contract_address: str, name: str) -> Optional[models.Entrypoint]:
+    entrypoints = get_entrypoints(db, contract_address)
+    entrypoint = list(filter(lambda e: e.name == name, entrypoints))
+    if len(entrypoint) == 0:
+        return None
+    return entrypoint[0]
+
+def create_contract(db: Session, contract: schemas.ContractCreation):
+    c = {k:v for k,v in contract.model_dump().items() if k not in ['entrypoints']}
+    db_contract = models.Contract(**c)
+    db.add(db_contract)
+    db.commit()
+    db.refresh(db_contract)
```

Comment:
I'm not sure how sessions work but I don't think it's a good idea to have two separate commits here. I'd rather insert the contract and its entrypoints in one transaction.

Reply:
Yes, It was hard to have something that works but I'll put a TODO on this function to rewrite it with transaction or something else better